common/app/model/dotcomrendering/DotcomRenderingUtils.scala

```diff
@@ -241,19 +242,27 @@ object DotcomRenderingUtils {
     }
   }
 
-  def shouldAddAffiliateLinks(content: ContentType): Boolean = {
-    AffiliateLinksCleaner.shouldAddAffiliateLinks(
-      switchedOn = Switches.AffiliateLinks.isSwitchedOn,
-      section = content.metadata.sectionId,
-      showAffiliateLinks = content.content.fields.showAffiliateLinks,
-      supportedSections = Configuration.affiliateLinks.affiliateLinkSections,
-      defaultOffTags = Configuration.affiliateLinks.defaultOffTags,
-      alwaysOffTags = Configuration.affiliateLinks.alwaysOffTags,
-      tagPaths = content.content.tags.tags.map(_.id),
-      firstPublishedDate = content.content.fields.firstPublicationDate,
-      pageUrl = content.metadata.id,
-      contentType = "article",
-    )
+  def shouldAddAffiliateLinks(content: ContentType)(implicit request: RequestHeader): Boolean = {
+    val contentHtml = Jsoup.parse(content.fields.body)
+    val bodyElements = contentHtml.select("body").first().children()
```

Comment:
I'm not wedded to this suggestion, and I'm not sure of the best place, but would an explanatory comment help?
```suggestion
    val bodyElements = contentHtml.select("body").first().children()
    /**
     * On smaller devices, the disclaimer is inserted before paragraph 2 of the article body and floats left.
     * This logic ensures there are two clear paragraphs of text at the top of the article.
     * We don't support inserting the disclaimer next to other element types.
     * It also ensures the second paragraph is long enough to accommodate the disclaimer appearing alongside it.
     */
    
```